### PR TITLE
Add setting BMC roles to upgrade instructions

### PR DESCRIPTION
**Pull request by alvarez3-hpe**
## Summary and Scope

Add instructions for verifying that BMCs of NCNs have the Management role in HSM and setting them as Management if they are not.

## Issues and Related PRs

* Resolves [CASMINST-4503](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4503)

### Tested on:

  * loki

### Test description:

Manually tested the commands that were added on loki 

## Risks and Mitigations

None

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by rsjostrand-hpe — APPROVED** [review]

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
The same changes as from the main branch PR are needed here:
https://github.com/Cray-HPE/docs-csm/pull/1430

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 1]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+    "x3000c0s5b0 Role=Management"
+    ```
+
+1. Run the following command to ensure they're all locked (NCNs and their BMCs).
+
+    ```
+    ncn-m001# /opt/cray/csm/scripts/admin_access/lock_management_nodes.py
+    ```
+
+For more information about setting NCN BMC roles see [Set BMC Management Role](../../operations/hardware_state_manager/Set_BMC_Management_Role.md)
```
Comment (nrockershousen) [inline upgrade/1.2/Stage_0_Prerequisites.md:248]:
```suggestion
For more information about setting NCN BMC roles, see [Set BMC Management Role](../../operations/hardware_state_manager/Set_BMC_Management_Role.md).
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 2]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+
+For more information about locking NCNs and their BMCs see [Lock and Unlock Nodes](../../operations/hardware_state_manager/Lock_and_Unlock_Management_Nodes.md)
```
Comment (nrockershousen) [inline upgrade/1.2/Stage_0_Prerequisites.md:250]:
```suggestion
For more information about locking NCNs and their BMCs, see [Lock and Unlock Nodes](../../operations/hardware_state_manager/Lock_and_Unlock_Management_Nodes.md).
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 3]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+    "x3000c0s9b0 Role=Management"
+    "x3000c0s2b0 Role=Management"
+    "x3000c0s6b0 Role=Management"
+    "x3000c0s3b0 Role=Management"
+    "x3000c0s7b0 Role=Management"
+    "x3000c0s4b0 Role=Management"
+    "x3000c0s5b0 Role=Management"
+    ```
+
+1. Run the following command to ensure they're all locked (NCNs and their BMCs).
```
Comment (nrockershousen) [inline upgrade/1.2/Stage_0_Prerequisites.md:242]:
```suggestion
1. Ensure the NCNs and their BMCs are all locked.
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 4]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
 
 To prevent any possibility of losing Workload Manager configuration data or files, a back-up is required. Please execute all Backup procedures (for the Workload Manager in use) located in the `Troubleshooting and Administrative Tasks` sub-section of the `Install a Workload Manager` section of the `HPE Cray Programming Environment Installation Guide: CSM on HPE Cray EX`. The resulting backup data should be stored in a safe location off of the system.
 
+<a name="lock-ncn-bmcs"></a>
+
+## Stage 0.6 - Verify NCN BMC Roles and Locks
+
+Both NCNs and their BMCs require their roles to be set to 'Management' in HSM for admin locking purposes. This is done automatically for NCNs but manually for their BMCs. For BMCs, this role may not have been set during the previous installation.
+
+1. Run the following command to ensure that the roles are properly set.
```
Comment (nrockershousen) [inline upgrade/1.2/Stage_0_Prerequisites.md:227]:
```suggestion
1. Ensure that the roles in HSM are properly set.
```

**Review by nrockershousen — CHANGES_REQUESTED** [review]

**Review by nrockershousen — APPROVED** [review]

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**